utils/daos_pylint.py

```diff
@@ -0,0 +1,249 @@
+        if self.scons_files:
+            desc += f'scons files: {",".join(self.scons_files)}\n'
+        return desc
+
+    def run(self, args):
+        """Run pylint against all files"""
+        print(self)
+        if self.files:
+            parse_file(args, self.files)
+        if self.ftest_files:
+            parse_file(args, self.ftest_files, ftest=True)
+        if self.scons_files:
+            for file in self.scons_files:
+                parse_file(args, file, scons=True)
+
+
+def parse_file(args, target_file, ftest=False, scons=False):
+    """Main program"""
+
+    rep = CollectingReporter()
+
+    wrapper = None
+
+    init_hook = None
+
+    if isinstance(target_file, list):
+        target = list(target_file)
+        target.extend(['--jobs', str(min(len(target_file), 20))])
+    elif scons:
+        wrapper = sl.check_script.WrapScript(target_file, output=f'{target_file}.pycheck')
+        target = [wrapper.wrap_file]
+        # Do not warn on module name for SConstruct files, we don't get to pick their name.
+        target.extend(['--disable', 'invalid-name'])
+        init_hook = """import sys
+sys.path.append('site_scons')
+sys.path.insert(0, 'utils/sl/fake_scons')"""
+    else:
+        target = [target_file]
+
+    if ftest:
+        target.extend(['--disable', 'consider-using-f-string'])
+        init_hook = """import sys
+sys.path.append('src/tests/ftest')
+sys.path.append('src/tests/ftest/util/apricot')
+sys.path.append('src/tests/ftest/cart/util/')
+sys.path.append('src/tests/ftest/util')
+sys.path.append('src/client')
+sys.path.append('site_scons')"""
+
+    target.extend(['--persistent', 'n'])
+    if init_hook:
+        target.extend(['--init-hook', init_hook])
+
+    if args.rcfile:
+        target.extend(['--rcfile', args.rcfile])
+
+    results = Run(target, reporter=rep, do_exit=False)
+
+    types = Counter()
+    symbols = Counter()
+
+    for msg in results.linter.reporter.messages:
+        vals = {}
+        # Spelling mistake, do not complain about message tags.
+        if msg.msg_id in ('C0401', 'C0402'):
+            if ":avocado:" in msg.msg:
+                continue
+        # Inserting code can cause wrong-module-order.
+        if scons and msg.msg_id == 'C0411' and 'from SCons.Script import' in msg.msg:
+            continue
+
+        if wrapper:
+            vals['path'] = target_file
+            vals['line'] = wrapper.convert_line(msg.line)
+        else:
+            vals['path'] = msg.path
+            vals['line'] = msg.line
+        vals['column'] = msg.column
+        vals['message-id'] = msg.msg_id
+        vals['message'] = msg.msg
+        vals['symbol'] = msg.symbol
+
+        # Duplicates, needed for message_template.
+        vals['msg'] = msg.msg
+        vals['msg_id'] = msg.msg_id
+        vals['category'] = msg.category
+
+        category = msg.category
+        # The build/scons code is mostly clean, so only allow f-string warnings.
```

Comment:
f-string warnings should be disabled now, right?

Reply:
No.  This is a (almost) complete re-write, so it uses a different pylintrc and f-string warnings are still enabled.  The intent here is to have some checks we run on the tree and expect to pass and have other checks we run on the diff, which would include the f-string code, so this PR does fail or fix f-strings but should help nudge changes towards them.